### ISSUE TITLE
Fail to link .so lib

### ISSUE THREAD

**Issue by overlook1234** [asdfqwrasdf/ArcFaceDemo#35]
When I try to integrate your sdk with my project for the face detection functionality, it throws exception at the line

`    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_face_detect);   // crash here
    }`

The logcat trace indicates that the something wrong in librender.so.

FaceDetectActivity.java
```
package cn.com.hotcent.eyecare.activity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.graphics.ImageFormat;
import android.graphics.YuvImage;
import android.hardware.Camera;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.arcsoft.facedetection.AFD_FSDKEngine;
import com.arcsoft.facedetection.AFD_FSDKError;
import com.arcsoft.facedetection.AFD_FSDKFace;
import com.guo.android_extend.java.AbsLoop;
import com.guo.android_extend.java.ExtByteArrayOutputStream;
import com.guo.android_extend.widget.CameraFrameData;
import com.guo.android_extend.widget.CameraGLSurfaceView;
import com.guo.android_extend.widget.CameraSurfaceView;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import cn.com.hotcent.eyecare.R;

public class FaceDetectActivity extends AppCompatActivity implements CameraSurfaceView.OnCameraListener, View.OnTouchListener, Camera.AutoFocusCallback{
    private final String TAG = this.getClass().getSimpleName();
    private static final String APPID = "H3Qa8ho6J9V3hhgfWmVYjWbBBDPUmGCb1xYvB76AEf4N";
    private static final String FD_KEY = "2rBiHTjU6tiJzxuDTeJjGvhtN7fBBWAyMQHV38dcdiGh";

    private int mWidth, mHeight, mFormat;
    private CameraSurfaceView mSurfaceView;
    private CameraGLSurfaceView mGLSurfaceView;
    private Camera mCamera;

//    AFT_FSDKVersion version = new AFT_FSDKVersion();
//    AFT_FSDKEngine engine = new AFT_FSDKEngine();
//    ASAE_FSDKVersion mAgeVersion = new ASAE_FSDKVersion();
//    ASAE_FSDKEngine mAgeEngine = new ASAE_FSDKEngine();
//    ASGE_FSDKVersion mGenderVersion = new ASGE_FSDKVersion();
//    ASGE_FSDKEngine mGenderEngine = new ASGE_FSDKEngine();
//    List<AFT_FSDKFace> result = new ArrayList<>();
//    List<ASAE_FSDKAge> ages = new ArrayList<>();
//    List<ASGE_FSDKGender> genders = new ArrayList<>();

    int mCameraID;
    int mCameraRotate;
    boolean mCameraMirror;
    byte[] mImageNV21 = null;
//    FRAbsLoop mFRAbsLoop = null;
//    AFT_FSDKFace mAFT_FSDKFace = null;
    Handler mHandler;

    Runnable hide = new Runnable() {
        @Override
        public void run() {
            mTextView.setAlpha(0.5f);
            mImageView.setImageAlpha(128);
        }
    };



    private TextView mTextView;
    private TextView mTextView1;
    private ImageView mImageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_face_detect);
    }

    @Override
    public void onAutoFocus(boolean success, Camera camera) {

    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        return false;
    }

    @Override
    public Camera setupCamera() {
        return null;
    }

    @Override
    public void setupChanged(int format, int width, int height) {

    }

    @Override
    public boolean startPreviewLater() {
        return false;
    }

    @Override
    public Object onPreview(byte[] data, int width, int height, int format, long timestamp) {
        return null;
    }

    @Override
    public void onBeforeRender(CameraFrameData data) {

    }

    @Override
    public void onAfterRender(CameraFrameData data) {

    }
}

```

activity_face_detect.xml

`<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="cn.com.hotcent.eyecare.activity.FaceDetectActivity">

    <com.guo.android_extend.widget.CameraSurfaceView
        android:id="@+id/surfaceView"
        android:layout_width="1dp"
        android:layout_height="1dp"/>

    <com.guo.android_extend.widget.CameraGLSurfaceView
        android:id="@+id/glsurfaceView"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:layout_centerInParent="true"/>

    <ImageView
        android:id="@+id/imageView"
        android:layout_width="120dp"
        android:layout_height="120dp"
        android:layout_marginLeft="10dp"
        android:layout_marginTop="10dp"/>

    <TextView
        android:id="@+id/textView"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_alignLeft="@+id/imageView"
        android:layout_alignRight="@+id/imageView"
        android:layout_below="@+id/imageView"
        android:layout_marginTop="10dp"
        android:text="@string/app_name"
        android:textAlignment="center"/>

    <TextView
        android:id="@+id/textView1"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_alignLeft="@+id/imageView"
        android:layout_alignRight="@+id/imageView"
        android:layout_below="@+id/textView"
        android:layout_marginTop="10dp"
        android:text="@string/app_name"
        android:textAlignment="center"/>
</RelativeLayout>
`

Logcat trace
```

01-03 14:23:37.119 1286-1415/? I/ActivityManager: Displayed cn.com.hotcent.eyecare/.activity.FaceDetectActivity: +138ms
01-03 14:23:37.127 3657-4226/cn.com.hotcent.eyecare I/ATC.: image.so JNI_OnLoad
01-03 14:23:37.127 3657-4226/cn.com.hotcent.eyecare D/ATC.: glesInit() <--- format = 0
01-03 14:23:37.127 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called  shader = 1 GL_INVALID_ENUM = 1280 GL_INVALID_OPERATION = 1282
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called compiled = 1, shader = 1
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glAttachShader
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glBindAttribLocation
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glLinkProgram
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare E/ATC.: link failed
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glesInit() <--- format = 0
01-03 14:23:37.133 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called  shader = 2 GL_INVALID_ENUM = 1280 GL_INVALID_OPERATION = 1282
01-03 14:23:37.138 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called compiled = 1, shader = 2
01-03 14:23:37.138 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called  shader = 3 GL_INVALID_ENUM = 1280 GL_INVALID_OPERATION = 1282
01-03 14:23:37.140 3657-4226/cn.com.hotcent.eyecare D/ATC.: glGetShaderiv called compiled = 1, shader = 3
01-03 14:23:37.148 3657-4226/cn.com.hotcent.eyecare D/ATC.: glLinkProgram
01-03 14:23:37.148 3657-4226/cn.com.hotcent.eyecare D/ATC.: glesInit() --->
01-03 14:23:37.148 3657-4226/cn.com.hotcent.eyecare D/ATC.: glesChanged(720, 1120) <---
01-03 14:23:37.148 3657-4226/cn.com.hotcent.eyecare A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0xd0 in tid 4226 (GLThread 1384)
01-03 14:23:37.149 3657-3952/cn.com.hotcent.eyecare D/OpenGLRenderer: endAllStagingAnimators on 0xab92b750 (RippleDrawable) with handle 0xab8dae50
01-03 14:23:37.210 624-624/? A/DEBUG: pid: 3657, tid: 4226, name: GLThread 1384  >>> cn.com.hotcent.eyecare <<<
01-03 14:23:37.250 624-624/? A/DEBUG:     #00 pc 000029f2  /data/app/cn.com.hotcent.eyecare-1/lib/arm/librender.so (GLChanged+25)
01-03 14:23:37.250 624-624/? A/DEBUG:     #01 pc 0000196b  /data/app/cn.com.hotcent.eyecare-1/lib/arm/librender.so
```

**Comment by overlook1234** [comment]
activity_face_detect.xml
```

<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="cn.com.hotcent.eyecare.activity.FaceDetectActivity">

    <com.guo.android_extend.widget.CameraSurfaceView
        android:id="@+id/surfaceView"
        android:layout_width="1dp"
        android:layout_height="1dp"/>

    <com.guo.android_extend.widget.CameraGLSurfaceView
        android:id="@+id/glsurfaceView"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:layout_centerInParent="true"/>

    <ImageView
        android:id="@+id/imageView"
        android:layout_width="120dp"
        android:layout_height="120dp"
        android:layout_marginLeft="10dp"
        android:layout_marginTop="10dp"/>

    <TextView
        android:id="@+id/textView"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_alignLeft="@+id/imageView"
        android:layout_alignRight="@+id/imageView"
        android:layout_below="@+id/imageView"
        android:layout_marginTop="10dp"
        android:text="@string/app_name"
        android:textAlignment="center"/>

    <TextView
        android:id="@+id/textView1"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_alignLeft="@+id/imageView"
        android:layout_alignRight="@+id/imageView"
        android:layout_below="@+id/textView"
        android:layout_marginTop="10dp"
        android:text="@string/app_name"
        android:textAlignment="center"/>
</RelativeLayout>

```

**Comment by asdfqwrasdf** [comment]
this issue is already solved in [android-extend #4](https://github.com/gqjjqg/android-extend/issues/4)
